File: Shortest Word Distance.py
""""// Time Complexity : O(n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this :
"""
class Solution:
    def shortestDistance(self, wordsDict: List[str], word1: str, word2: str) -> int:
        p1 = -1
        p2 = -1
        minn = float('inf')
        for i in range(len(wordsDict)):
            if wordsDict[i] == word1:
                p1 = i
            elif wordsDict[i] == word2:
                p2 = i
            if p1 != -1 and p2 != -1:
                minn = min(minn, abs(p2-p1))
            if minn == 1:
                return minn
        return minn

File: shortest_word_distance.java
class Solution {
    public int shortestDistance(String[] wordsDict, String word1, String word2) {
        int i = -1;
        int j = -1;
        int min=Integer.MAX_VALUE;
        for(int a =0; a<wordsDict.length;a++){
            if(wordsDict[a].equals(word1)) i=a;
            if(wordsDict[a].equals(word2)) j=a;
            if(i!=-1&&j!=-1) min = Math.min(min,Math.abs(i-j));
        }
    return min;}
}